docs/specs/short-read-alignment-vector.md

```diff
@@ -0,0 +1,215 @@
+  the pipeline is launched for a given batch of samples. I.e., there
+  is no need to consider any case where additional lanelets may arrive
+  at a later time.
+
+* Sequencing will be from Illumina technology, using paired-end
+  sequencing. Mosquitoes will be sequenced individually, i.e., each
+  mosquito sample will be used to create a separate DNA library, and
+  libraries will not be pooled. Each individual mosquito will be
+  sequenced to high coverage (target ~30X). Read length may vary
+  between sequencing runs, i.e., the pipeline cannot assume the same
+  read length across all lanelets.
+
+* Trimming of adapter sequences, if necessary, will have been
+  performed prior to running this pipeline.
+
+
+## Inputs
+
+* Reference sequence - The reference genome sequence to align to, in
+  FASTA format.
+
+* Sequence reads - For each lanelet, a pair of files containing
+  sequence reads, with one file containing forward reads and one file
+  containing reverse reads, in FASTQ format.
+
+* Lanelet manifest - A tab delimited file with one row per
+  lanelet. Mandatory columns are as follows, may contain any other
+  columns:
+
+    * `read1_path` - URL or local filesystem path to FASTQ file
+      containing forward reads.
+
+    * `read2_path` - URL or local filesystem path to FASTQ file
+      containing reverse reads.
+
+    * `sample_id` - Identifier for the sample that was sequenced.
+
+
+## Outputs
+
+* One analysis-ready BAM file for each sample.
+
+* samtools stats output for each sample.
+
+* GATK CallableLoci output for each sample.
+
+
+## Software
+
+* bwa 0.7.15
+* samtools 1.4.1 (htslib 1.4.1)
+* picard 2.9.2
+* biobambam 2.0.73
+* GATK 3.7-0
+
+
+## Pipeline
+
+
+### Sub-pipeline: Lanelet alignment
+
+For each lanelet, the following steps are run.
+
+
+#### Step: Read alignment
+
+Align sequence reads for a single lanelet to the reference sequence with command:
+
+```
+bwa mem -M -T 0 -R readgroup_string ref.fa read1.fastq read2.fastq > output.sam
+```
+
+Notes:
+
+* -M marks split alignments as secondary (for Picard compatibility)
+
+* -T 0 sets the threshold for alignment score output to zero
+
+
+#### Step: Read alignment post-processing
+
+Conversion of sam to bam via namesort, fixmates, coordinate sort and
+calmd:
+
+```
+samtools view -bu output.sam 
+| samtools sort -n 
+| samtools fixmate 
+| samtools sort  
+| samtools calmd -b ref.fa > output.bam
+```
+
+
+### Sub-pipeline: Sample alignment merging and improvement
+
+For each sample, the following steps are run.
+
+
+#### Merge to sample
+
+Merge alignments from all lanes to build a BAM/CRAM for a single sample:
+
+```
+picard.jar MergeSamFiles lane1.bam lane2.bam lane3.bam 
+```
+
+
+#### Mark duplicate reads
+
+Mark duplicate reads with command `bammarkduplicates` from biobambam.
+
+
+#### Indel realignment
+
+Realign reads around indels found within the alignments using the following commands:
+
+```
+GATK -T RealignerTargetCreator
+```
+
+No known indels.
```

Comment:
What do you mean by this?

Reply:
Ditto that question - if there are no known indels why do we need this task?

Reply:
IIRC the RealignerTargetCreator tool scans along the genome and looks for patterns in the alignments which are suggestive of indel realignment errors. I can't remember the details, but it's something like looking for clusters of base mismatches. Any such region gets output as a target region where the next step will perform indel realignment. 

Additionally if you know ahead of time of any indels that have been found previously, IIRC you can input these to RealignerTargetCreator, and I guess it will either just automatically output those regions as targets, or may do something like look at those regions more carefully for signatures of indel errors. 

But in our case we don't have any prior knowledge of indels, so we just rely on RealignerTargetCreator outputting targets based on it's detection of indel alignment error signatures in the alignments.